### ISSUE TITLE
Running on Wine in Linux not usable

### ISSUE THREAD

**Issue by knuxyl** [project64/project64#1234]
Plugins and Settings page will not load on Wine. This new settings layout is also very glitchy on Windows. I think going back to the original layout which didn't require whatever crazy requirements this page has. I mean is this .NET, C++ SP6? What libraries or runtime are required?

**Comment by project64** [comment]
the change on windows was needed to proper support unicode for non eglish languages. 

where is the case where they are glitchy on windows?

The core of pj64 is now linux compatible, be better to see a native version for linux be created

**Comment by oddMLan** [comment]
This is Wine's bug, not PJ64
duplicate #373

**Comment by knuxyl** [comment]
How can I compile this on Linux then? I am running a linux machine, and I see no cmake or make files.

**Comment by LegendOfDragoon** [comment]
>How can I compile this on Linux then?

Dunno, but it's not fully ported to Linux yet, afaik.

**Comment by cxd4** [comment]
>the change on windows was needed to proper support unicode for non eglish languages.

Could we have **English** languages instead of _"eglish"_ ones and have all non-English folk to go Unicode themselves in the UTF.

Also, I can't test on WINE because I don't have 32-bit WINE.  The distro is 64-bit and I don't know how to put a 32-bit WINE with the multi libs all on it, so I'd rather wait until I can find a 64-bit build of Project64 somewhere.  Anyone have a link to a 64-bit compile of Project64 (using the interpreter of course since IIRC there is no 64-bit dynarec yet)?

**Comment by cxd4** [comment]
>Plugins and Settings page will not load on Wine. This new settings layout is also very glitchy on Windows.

How does the glitchiness of the Plugins and Settings page make Project64 unusable on any OS?

Use the defaults or if you have to edit the config file I would imagine, if the GUI is that bugged.  But if the layout is wrong you can still tab through it and use the GUI anyway even if the text isn't formatted nicely.

**Comment by knuxyl** [comment]
@cxd4 
emucr.com has the best up to date builds. ton of different emulators.
And I have 64 bit linux too with wine. Wine can run as either 32 or 64 bit on a 64 bit installation.
Do this
`WINEPREFIX="$HOME/prefix32" WINEARCH=win32 wine wineboot`
This creates the environment for 32 bit. Then, if you are using winetricks to install dlls and such, do
`WINEPREFIX="$HOME/prefix32" WINEARCH=win32 winetricks`
And to launch programs in the 32 bit environment
`WINEPREFIX="$HOME/prefix32" WINEARCH=win32 wine PROGRAM`

**Comment by Frank-74** [comment]
> emucr.com has the best up to date builds.

Cough, splutter.  I'm not very well...

**Comment by cxd4** [comment]
None of that stuff works here.
```
bash-4.3$ WINEPREFIX="$HOME/.wine32" WINEARCH=win32 wine wineboot
bash: /usr/bin/wine: No such file or directory
bash-4.3$ WINEPREFIX="$HOME/.wine32" WINEARCH=win32 winetricks
bash: winetricks: command not found
bash-4.3$ WINEPREFIX="$HOME/.wine32" WINEARCH=win32 wine PROGRAM.EXE
bash: /usr/bin/wine: No such file or directory
```

The reason is that I do have `/usr/bin/wine64` as well as `/usr/bin/wine`, but the latter is a 32-bit executable which the command shell fails to recognize as a executable file because there are no 32-bit libs on this system.  I don't even have the 32-bit kernel Linux put on here because I just haven't had a reason to bother looking at setting up 32-bit compatibility here, up until maybe now.

Are you familiar with installing 32-bit versions of the OS libs to a Slackware machine? :)

>emucr.com has the best up to date builds.

It has ass.

**Comment by knuxyl** [comment]
@cxd4 
I don't understand why you would have the 32-bit binary on a 64-bit machine (even though the results from terminal are telling me otherwise....). The 64-bit binary can run in 32-bit mode.I'm familiar with Slackware which is why I do not use it. Their software repository system is crap. I recommend a Debian system like Xubuntu or Mint. Then when you try to install wine through the package system (apt) then it will automatically install all needed libraries. Again, you should not have 2 separate wine installations, you only need one, with there being 2 different environments. You need to start over because it's not saying it doesn't recognize wine as an executable, it's saying it's not where it should be. Those commands I gave you (except the second one, it's just an example) are the only things you need to setup and run in the 32-bit environment with the 64-bit installation of wine.

**Comment by cxd4** [comment]
>(even though the results from terminal are telling me otherwise....).

No, they're not.

I'll repeat.  The 32-bit executable is in fact on-disk, but it says "no such file or directory" only when trying to **execute** it because I have installed no 32-bit operating system components.

>I'm familiar with Slackware which is why I do not use it. Their software repository system is crap. I recommend a Debian system like Xubuntu or Mint.

Right...so, in other words, you don't know the solution. :)

Your "solution" is switch to a distribution that automates the process where no solution was really needed because it's done for you (whatever "it" is).

>You need to start over because it's not saying it doesn't recognize wine as an executable, it's saying it's not where it should be.

Whatever you think it's saying isn't the problem.

```cd /usr/bin/
vi ./wine # to open for reading ```

works,

```cd /usr/bin/
./wine # to execute it```

fails.

**Comment by theAntiBob** [comment]
perhaps i can help...
its not about your wine being 32v64 bit. or your os
when creating a wineprefix, you want the winearch variable to be set to win32
the only time your wineprefix (you really should isolate all of your windows programs and give them their own wineprefix instead of all of your programs residing in the .wine [default] prefix) should be 64 bit is when the software you are trying to run is a 64 bit program.

here is my methodology:

create a CLEAN (not used) wineprefix by typing running this short script at the command console:

user@machine ~/: `WINEPREFIX=~/project64 WINEARCH=win32 winecfg` 
^step 1 for the quicked

 WINEPREFIX <-- MUST NOT EXIST! your new wineprefix must not be an empty folder, it must not exist yet. it will be created
WINEARCH <-- This is the architecture you are specifying... win32, obviously this should be the default, i don't know what's wrong with those wine guys, why isn't it!?
winecfg <-- this is the command you are actually running with the WINEPREFIX and WINEARCH variables being temporarily set inline before your command winecfg ; you can EXPORT these variables for shell scripts (sh) or a terminal session... it's not so hard to type the 50th time, trust me...
winecfg will create a new wineprefix since WINEPREFIX doesn't exist (following along?)

after you have created your wineprefix, put your project64 in the virtual c drive (cp, thunar, however you want to copy it there is fine...) you're looking for a place like ~/project64/drive_c/

you probably have to launch project64 from the proper working directory, and to avoid having to type out picketfencefest, it's easier to just launch the wine filemanager and work from there

like this

user@machine ~/: `WINEPREFIX=~/project64 winefile`
^step 2 for the quicked
notice you don't need to specify WINEARCH? once a wineprefix has been created, it is either win32 or not. period.

go to your virtual c drive by clicking on the C: in the drive bar. double click on `Setup\ Project64\ v2.-3-210-g4f0ca48.exe` <-- i'm putting the slashspaces just because spaces aren't good in names for *nix filesystems or ANYONE else for that matter, if you don't know now you know
i'm not going to discuss security issues resulting or derived from or related or pertaining to executing untrustable binaries from root directories... but you sure can if you feel like it. i just won't be around to hear it.

just use all the default locations things want to be installed... [oldmanrambleshere]i remember back in my m$ days i had to reformat every month or so, kept all my programs installed to the D: drive, still had to reinstall everything because none of the registry keys were there after kicking windows off the C: drive for eating all the available 10gb i would ever leave around for it to puke itself all over...[this continues on for several years before the discovery of WINEPREFIX and an extensive collection of wine"bottles"]

after the installation process has completed you can launch project64 from winefile (as long as you launch it with WINEPREFIX set properly, launch it from the programs menu (under wine/programs) or navigate to the project64 install directory and launch it directly, something like this:

user@machine ~/: `cd ~/project64/drive_c/Program\ Files/Project64\ 2.3/ ; WINEPREFIX=~/project64 wine "C:\Program Files\Project64 2.3\Project64.exe"`
^step 3 for the quicked

the \" or "quotes" ARE necessary.

have fun.

**Comment by knuxyl** [comment]
@cxd4 

I recommend you uninstall your linux and install Windows.

Back on topic, how can I compile this for Linux??

**Comment by theAntiBob** [comment]
that's what i would like to know as well.

just thought i would clarify the 32/64 wine situation for the other folks

**Comment by Jj0YzL5nvJ** [comment]
Check for help here: https://www.winehq.org/download

When I install **winehq-devel** on my Xubuntu, the dependencies / prerequisites were these:
`gcc-5-base:i386 gcc-6-base:i386 gstreamer1.0-plugins-base:i386 libasn1-8-heimdal:i386 libasound2:i386 libasound2-plugins:i386 libasyncns0:i386 libavahi-client3:i386 libavahi-common-data:i386 libavahi-common3:i386 libbsd0:i386 libc6:i386 libcap2:i386 libcapi20-3 libcapi20-3:i386 libcdparanoia0:i386 libcomerr2:i386 libcups2:i386 libdb5.3:i386 libdbus-1-3:i386 libdrm-amdgpu1:i386 libdrm-intel1:i386 libdrm-nouveau2:i386 libdrm-radeon1:i386 libdrm2:i386 libedit2:i386 libelf1:i386 libexif12:i386 libexpat1:i386 libffi6:i386 libflac8:i386 libfontconfig1:i386 libfreetype6:i386 libgcc1:i386 libgcrypt20:i386 libgd3:i386 libgl1-mesa-dri:i386 libgl1-mesa-glx:i386 libglapi-mesa:i386 libglib2.0-0:i386 libglu1-mesa:i386 libgmp10:i386 libgnutls30:i386 libgpg-error0:i386 libgphoto2-6:i386 libgphoto2-port12:i386 libgpm2:i386 libgsm1:i386 libgssapi-krb5-2:i386 libgssapi3-heimdal:i386 libgstreamer-plugins-base1.0-0:i386 libgstreamer1.0-0:i386 libhcrypto4-heimdal:i386 libheimbase1-heimdal:i386 libheimntlm0-heimdal:i386 libhogweed4:i386 libhx509-5-heimdal:i386 libice6:i386 libicu55:i386 libidn11:i386 libieee1284-3:i386 libjack-jackd2-0:i386 libjbig0:i386 libjpeg-turbo8:i386 libjpeg8:i386 libjson-c2:i386 libk5crypto3:i386 libkeyutils1:i386 libkrb5-26-heimdal:i386 libkrb5-3:i386 libkrb5support0:i386 liblcms2-2:i386 libldap-2.4-2:i386 libllvm3.8:i386 libltdl7:i386 liblzma5:i386 libmpg123-0 libmpg123-0:i386 libncurses5:i386 libnettle6:i386 libodbc1 libodbc1:i386 libogg0:i386 libopenal1:i386 libopus0:i386 liborc-0.4-0:i386 libosmesa6 libosmesa6:i386 libp11-kit0:i386 libpcap0.8:i386 libpciaccess0:i386 libpcre3:i386 libpng12-0:i386 libpulse0:i386 libroken18-heimdal:i386 libsamplerate0:i386 libsane:i386 libsasl2-2:i386 libsasl2-modules:i386 libsasl2-modules-db:i386 libselinux1:i386 libsm6:i386 libsndfile1:i386 libspeexdsp1:i386 libsqlite3-0:i386 libssl1.0.0:i386 libstdc++6:i386 libsystemd0:i386 libtasn1-6:i386 libtheora0:i386 libtiff5:i386 libtinfo5:i386 libtxc-dxtn-s2tc0:i386 libudev1:i386 libusb-1.0-0:i386 libuuid1:i386 libv4l-0:i386 libv4lconvert0:i386 libvisual-0.4-0:i386 libvorbis0a:i386 libvorbisenc2:i386 libvpx3:i386 libwind0-heimdal:i386 libwrap0:i386 libx11-6:i386 libx11-xcb1:i386 libxau6:i386 libxcb-dri2-0:i386 libxcb-dri3-0:i386 libxcb-glx0:i386 libxcb-present0:i386 libxcb-sync1:i386 libxcb1:i386 libxcomposite1:i386 libxcursor1:i386 libxdamage1:i386 libxdmcp6:i386 libxext6:i386 libxfixes3:i386 libxi6:i386 libxinerama1:i386 libxml2:i386 libxpm4:i386 libxrandr2:i386 libxrender1:i386 libxshmfence1:i386 libxslt1.1:i386 libxt6:i386 libxxf86vm1:i386 wine-devel wine-devel-amd64 wine-devel-i386:i386 winehq-devel zlib1g:i386`

I hope it serves as a reference.

**Comment by cxd4** [comment]
@theAntiBob thanks for the tips.

I know there are a number of solutions to this, but I haven't had time these days to do much of anything on the computer (at least while getting wi-fi).

Getting Win32 32-bit to work in WINE is not time-consuming; I just lack free time very much.

If I didn't lack said time I'd be calling people out on these repo threads every day, not every week. :)

**Comment by cxd4** [comment]
@knuxyl , isn't it partially redundant to suggest I install Windows, when you've already recommended switching to Mint Linux and Ubuntu? :P

With n00b distros like those you're hardly qualified to comment on Slackware's packaging system.

And I didn't install their i386 package.  I took the x64 package for Slackware from their official site:
https://sourceforge.net/projects/wine/files/Slackware%20Packages/

>EDIT try winecfg and see what that does

Am not on the same laptop as previously so cannot test yet.  Busy with life.

At any rate, with or without adding a i386 package for wine, it doesn't work out of the box.  I know I can **fiddle** with it to get it to work in a number of ways but thought it was easier to ask people here for a mirror of a 64-bit build of Project64 for Windows, since Win64 stuff already works in my WINE by executing `wine64 [prog]`.  Wiping Linux and installing Windows would be the most time-consuming and ridiculously inefficient option; why don't you wipe your Windows and install DOS?

>Back on topic, how can I compile this for Linux??

By compiling it Unix-style, using the scripts I'd added.

You'd get the bonus of the compiled and linked software actually running (and playing N64 games) if zilmar wasn't on the devoid side of attempting a little communication with my efforts.

**Comment by knuxyl** [comment]
@cxd4 
i use xubuntu because I'm lazy. I have experience with Slackware... it's a pain in the ass. I'd rather not have to basically program my operating system, I'll let others do that. I'm just saying, a debian based distribution will be more friendly, a much bigger community, more packages, more help, etc. All linux is the same at the core, the kernel, and I studied which one is the most popular and efficient, and came up with debian based with a lightweight desktop, ie xubuntu or lubuntu. lubuntu works crap on my pc, but it is probably better. xubuntu is prettier and all i have to do is sudo apt-get install winehq-devel and boom, playing world of warcraft, unreal tournament 2014, counter strike 1.6, and civilization v with no problems. it doesn't matter if you're trying to get Wine working so you can use latest builds of Project64. They do not work. They will not initialize the plugins and the settings right side panel does not load

**Comment by cxd4** [comment]
My order went I guess like Knoppix, Backtrack, Debian, Lubuntu, Parabola, Xubuntu, Fedora, Slackware.

So Xubuntu was what I was using before I wiped the whole laptop for a single distro.  I would tend to recommend it to practical people who need their pragmatism before all else, but I will not switch back to it.

Lubuntu I just found visually unattractive...mostly because it came with Leafpad, not Mousepad.

>All linux is the same at the core, the kernel,

That is because by the time you're experienced with something like Slackware, all distributions start to seem like the exact same thing with different names, just shipped with different default skins and such.  Or package managers...but, package installation isn't really part of the design as a Unix.

>xubuntu is prettier and all i have to do is sudo apt-get install winehq-devel and boom, playing world of warcraft, unreal tournament 2014, counter strike 1.6, and civilization v with no problems.

I'm sure you get plenty of dependency bloat from the resolution Debian/Ubuntu has to do to get it to work, too. :) Unless you want to do the intelligent thing and allocate the dependency resolution manually on your own, which by default Slackware has you do, but this is not a permanent thing.

WINE is not something I favor having or keeping installed, so I wouldn't really prefer my distribution of choice to come with it ready to go as a package, nor do I need to demand `apt` voodoo to have it.


>it doesn't matter if you're trying to get Wine working so you can use latest builds of Project64. They do not work. They will not initialize the plugins and the settings right side panel does not load

...Or you're just lazy.

Anyway, I'm the one who ported Project64 to Linux in the first place, so I think I know how to manage switching to cleaner plugins than the ones Project64 comes with, and get emulation to work.

It really would be to contribute while being able to test pj64, not to simply "use" it as you suggest.

Right side of a settings panel has nothing to do with whether the emulator works.  If you have to configure it for some reason just edit the config files (again, something Slackware has you in the habit of doing :-)), or just compile changes with your config hard-coded.

**Comment by cxd4** [comment]
Here, see?
https://docs.slackware.com/slackware:multilib#advantage_of_a_multilib_system

_"Most Windows programs are still 32bit, and in order to run those on Linux with Wine, you need a 32bit version of Wine."_

64-bit WINE does not run 32-bit Windows programs without having the 32-bit Linux libraries installed on your 64-bit system.  It's the same way with Windows; you can't run 32-bit Windows programs on 64-bit Windows without having the 32-bit DLLs in C:/windows/syswow64.

I know from experience in using Xubuntu that Xubuntu ships with the 32-bit libraries already included (except the 32-bit libstdc++ for some reason), which is probably why you were fooled. :) But the fact that Slackware doesn't ship them by default (unless of course installing the 32-bit distro) is, again, not a disadvantage like you think; it's insurance that the distribution is set up and fully controlled to the way we need.  Because for years now I've never needed this 32-bit crap anyway.

**Comment by cxd4** [comment]
http://i.imgur.com/6AQKSWo.png

It installs and runs fine when you set the right plugins.

For most of the ROMs with default plugins though Project64 says some memory exception I didn't take the time to screenshot.  I uninstalled wine by half-accident anyway just by deleting the 32-bit packages I had to install to get it to work.  WINE is some garbage when projects should be written to be portable, and I don't mean C++ recreate everything for every single OS "portable".

**Comment by cxd4** [comment]
Your quote looks like you're asking why I evade Ubuntu, not why is (X distro here) the best.

I wiped Xubuntu partly because I got tired of having so many partitions.
https://gist.github.com/cxd4/c8a05356f189ff1b41f5420837037f3d

I would still recommend it to people learning to transition towards the better change away from Windows, but it is not really a Unix-like thing.  It's a modernist's thing.  I liked Debian more.

I had both Lubuntu and Xubuntu @LegendOfDragoon but think I accidentally linked you to just plain "Ubuntu".  (Xubuntu = Xfce (desktop & software suite), Lubuntu = Lxde and Ubuntu is just Ubuntu with Unity which is its own thing.)  It's all the same, just different skins, packages and stuff shipped with it out of the box.  That's why I'm saying the distributions themselves, even outside the Ubuntu family, aren't really that different in any permanent way at all once you're used to something like Slackware.

And I learned a lot by switching to something more breakable and hazard-prone to human fault and error.  A lot of these things I think are not possible to control with Ubuntu; normally Ubuntu is used as a Windows shield to learning Linux while avoiding Windows which again is progress, just insufficient for me.

**Comment by cxd4** [comment]
I have figured out how to set `Project64.cfg` manually to start emulation with initialized plugins.

```
[Plugin]
Audio Dll=No Sound.dll
Audio Dll Ver=No Sound
Controller Dll=keyboard.dll
Controller Dll Ver=System Keyboard
Graphics Dll=CFB.dll
Graphics Dll Ver=Basic CFB plugin by zilmar
RSP Dll=rsp.dll
RSP Dll Ver=Static Interpreter


[default]
Current Language=English

[Support Project64]
Run Count=41

[Recent File]
Recent Rom 0=Z:\n64\demo\SP_CRAP.V64
Recent Rom 1=Z:\n64\NBME_000.N64
Recent Rom 2=Z:\n64\NGXE_000.N64



[2E7E893C-4E68B642-C:0]
Good Name=Absolute Crap Intro 2 by Lem (PD)
32bit=0
Audio Signal=1
Counter Factor=1
CpuType=1
Delay DP=0
Fixed Audio=0
FuncFind=2
Linking=0
RDRamSize=8388608
SP Hack=0
Sync Audio=0
```

**Comment by cxd4** [comment]
Here is the result:
http://i.imgur.com/v1WmmwH.png

But I barely figured it out.  It was actually a black screen until I clicked off of Project64 thinking nothing I did would ever get it to work, then clicked back on and the emulation of DirectDraw in zilmar's CFB plugin finally kicked in and the window resized with the working frame buffer.

**Comment by cxd4** [comment]
...Actually scratch that CFG.  Default settings work under WINE, dynamic recompiler included.

In fact, Gauntlet Legends works in LLE under WINE.

http://i.imgur.com/rd7CDtd.png

Looks like this issue can be closed :P ... or at least renamed drastically.  The only thing about Project64 that doesn't seem to work under WINE is the fact the entire GUI is broken for configuring any of the settings, so you have to touch `Project64.cfg` like I did to change default plugins and such.  That being said, editing text files without a GUI practically constitutes a fraction of what it took to install the distribution of Linux I'm on in the first place, so that's not realistically an issue truly hindering usablity of the emulator.

**Comment by knuxyl** [comment]
When I was testing this on wine when I opened this, the plugins weren't loaded, which is why I was trying to get the settings page to load. I remember trying to set the configuration file manually, but it still wouldn't load the plugins. When I tried to start a game it would say "No Graphics plugin selected" or something along those lines. I'll test it again sooner or later and post back.

**Comment by cxd4** [comment]
That x Dll ver stuff is pretty useless, IIRC.  I think just the file name by its self causes Project64 to auto-fill the rest in to the CFG, and you don't have to type out the plugin name.

So really just
```
Audio Dll=No Sound.dll
Controller Dll=keyboard.dll
Graphics Dll=CFB.dll
RSP Dll=rsp.dll
```

You do have to have the DLLs in the root of the `Plugin/` directory for the above to work, though.

And, obviously, you'll have to use file names differently than those, because most of those DLL files are plugins that only a person like me keeps archived, except the No Sound plugin by zilmar.  You could try the DirectSound plugin by Jabo...I think it initializes fine under WINE but then changes its mind mid-way when emulation starts and switches to saying DirectSound can't get a device (even though VBA-M for Windows works fine under WINE here when I switch to DirectSound mode from XAudio).  I have not bothered to test different audio plugins.

**Comment by Frank-74** [comment]
On Windows, the default plugin section in the .cfg is this:

[Plugin]
Audio Dll Ver=Jabo's DirectSound 1.7.0.7
Controller Dll Ver=N-Rage For PJ64: 2.5.3.9999
Graphics Dll Ver=Jabo's Direct3D8 1.7.0.57-ver5
RSP Dll Ver=RSP Plugin 1.7.4.9999

Plugins can be in anywhere under the Plugin directory.
When you change the above defaults, the filename is written into the config as well:

Graphics Dll=GFX\Project64-Video.dll
Graphics Dll Ver=Project64 Video Plugin: 2.2.0.9999

**Comment by cxd4** [comment]
Most of those plugins don't work under WINE.  I don't know what N-Rage's problem is...XInput crap?

I don't have WINE configured for later Windows versions anyway.

>Plugins can be in anywhere under the Plugin directory.

Only if you add the "GFX\" Project64-Video.dll like you did, yes, but Graphics Dll = GFX\anything looks redundant.

@Frank-74 how do I disable that stupid Pause emulation when window is inactive?

And what's the CFG file code for turning off hiding of advanced settings (which really aren't that advanced at all...)?

**Comment by cxd4** [comment]
Only using the "Ver" lines doesn't work, by the way.

I only got plugins to initialize by using the DLL file names.  Saying something like "Audio Dll Ver" just reverts back to the core's embedded default of trying to reach Jabo's plugins.

**Comment by Frank-74** [comment]
@cxd4 
```
[default]
Auto Sleep=0
Basic Mode=0
```
The default plugins don't include the path/filename, just the Dll Ver name.

**Comment by cxd4** [comment]
Thanks.

>The default plugins don't include the path/filename, just the Dll Ver name.

So I had noticed.

Just one more ironic thing that threw me the heck off with trying to get the blasted CFG to do what I said.  My point still stands for any plugins that Project64 would consider "third-party"--the GFX/ etc. is needed in a full path.

**Comment by Frank-74** [comment]
XInput and XAudio work under XP SP3 with the last directx9 redist.  Maybe you need to install the DirectX9 redist under Wine to get it to work?  I don't know.

**Comment by cxd4** [comment]
It's not really something I investigated.  I just know I had DirectSound working for VBA-M, not Jabo.

http://i.imgur.com/BurTD1z.png

Haven't figured out how to use `dxdiag` from the command line or looked up how to do so.  I do see XAudio and XInput installed to the WINE system, though, and the plugins did initially load.  I would have to rearrange the stupid DLLs back to the way everything was to test Jabo's plugins again, and I had a hard enough time getting Project64 to not use them in the first place.

The most important thing is I got the classic UNIX CDE desktop to work with Project64. :P

**Comment by cxd4** [comment]
OK, tried Jabo's Direct3D8 1.7 again, and that does work.

In the Direct3D context ensuing from Jabo's plugin supplied with Project64 2.3, my video card is falsely detected to be an NVIDIA GeForce FX 5800 when I have actually a 6150 LE.

Might have something to do why z64gl is like ten times faster than Jabo's plugin under WINE.

Not testing any HLE video plugins.

**Comment by cxd4** [comment]
As for Jabo's DirectSound...that one still does not work ever.

At least with N-Rage Input (which doesn't work under WINE either) you can configure the DLL and get the GUI for configuring the plugin to open.  DirectSound plugin won't even make it that far.
```
bash-4.3$ wine ./Project64.exe 
preloader: Warning: failed to reserve range 00010000-00110000
preloader: Warning: failed to reserve range 00010000-00110000
fixme:resource:DestroyAcceleratorTable other process handle 0x1?
ALSA lib pcm_dsnoop.c:589:(snd_pcm_dsnoop_open) unable to create IPC shm instance
ALSA lib pcm_dmix.c:1011:(snd_pcm_dmix_open) unable to create IPC shm instance
ALSA lib pcm_mmap.c:341:(snd_pcm_mmap) mmap failed: Cannot allocate memory
ALSA lib pcm_direct.c:1316:(snd1_pcm_direct_initialize_secondary_slave) unable to mmap channels
ALSA lib pcm_dmix.c:1098:(snd_pcm_dmix_open) unable to initialize slave
fixme:ntdll:FILE_GetNtStatus Converting errno 12 to STATUS_UNSUCCESSFUL
fixme:ntdll:FILE_GetNtStatus Converting errno 12 to STATUS_UNSUCCESSFUL
fixme:msvcrt:__clean_type_info_names_internal (0x1361e234) stub
```

**Comment by cxd4** [comment]
AziAudio.dll ... I don't know whether it's the DS8 or XA2 version but it loads fine.

No errors from WINE anyway.  Emulation works, but no sound.  None of the plugin options seem to make a difference.  I'm not surprised, though...only reason I used DirectSound on VBA-M was that XAudio2 didn't work anyway there.

That OpenAL audio plugin I wrote would come in handy right about now if I had finished working on it.

**Comment by cxd4** [comment]
Alright, I wasted peoples' time by posting any of that.

There wasn't anything to the fault of the plugins or Project64 keeping DirectSound from working in WINE.  I set `autospawn = no` in `etc/apulse/client.conf` and rebooted my machine.  I made sure that the audio daemon was accessed by a non-root user before touching it with root.  Project64 works with Jabo's DirectSound (and who knows how many other DirectSound plugins) under WINE.

In short, it was an error with PulseAudio's redirection to ALSA.  PulseAudio is some jacked-up high-level audio API that runs on top of usually ALSA (which is itself inside of Linux) but with support for bluetooth audio, Skype, and a bunch of other crap I know nothing about.

It's a shame because for so many years, Slackware was one of like maybe only 2 Linux distributions in existence that didn't link everything against the controversial PulseAudio library and throw that in to the distro.  Nowadays the only maintained distro I know that still refuses to adopt PulseAudio is Gentoo.
